pkg/operator/operator.go

```diff
@@ -168,31 +169,37 @@ func (o *Operator) Start(stop <-chan struct{}) error {
 // ensureDefaultIngressController creates the default ingresscontroller if it
 // doesn't already exist.
 func (o *Operator) ensureDefaultIngressController() error {
+	name := types.NamespacedName{Namespace: o.namespace, Name: manifests.DefaultIngressControllerName}
+	ic := &operatorv1.IngressController{}
+	if err := o.client.Get(context.TODO(), name, ic); err == nil {
+		return nil
+	} else if !errors.IsNotFound(err) {
+		return err
+	}
+	infraConfig := &configv1.Infrastructure{}
+	if err := o.client.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig); err != nil {
+		return err
+	}
 	// Set the replicas field to a non-nil value because otherwise its
 	// persisted value will be nil, which causes GETs on the /scale
 	// subresource to fail, which breaks the scaling client.  See also:
 	// https://github.com/kubernetes/kubernetes/pull/75210
 	//
 	// TODO: Set the replicas value to the number of workers.
-	two := int32(2)
-	ic := &operatorv1.IngressController{
+	replicas := int32(2)
+	if infraConfig.Status.HighAvailabilityMode == configv1.NoneHighAvailabilityMode {
```

Comment:
Note that https://github.com/openshift/api/pull/827 currently has the `SingleReplicaTopologyMode` constant instead of `NoneHighAvailabilityMode`. We can pick that up when 827 merges.